bve/src/parse/train_dat/sections/device.rs

```diff
@@ -0,0 +1,94 @@
+use bve_derive::{FromKVPSection, FromKVPValueEnumNumbers};
+
+#[derive(Debug, Default, Clone, PartialEq, FromKVPSection)]
+pub struct DeviceSection {
+    #[kvp(bare)]
+    ats: AtsAvailable,
+    #[kvp(bare)]
+    atc: AtcAvailable,
+    #[kvp(bare)]
+    eb: EbAvailable,
+    #[kvp(bare)]
+    const_speed: ConstSpeedAvailable,
+    #[kvp(bare)]
+    hold_brake: HoldBrakeAvailable,
+    #[kvp(bare)]
+    re_adhesion_device: ReAdhesionDeviceType,
+    #[kvp(bare)]
+    load_compensating_device: String,
+    #[kvp(bare)]
+    pass_alarm: PassAlarmType,
+    #[kvp(bare)]
+    door_open_mode: DoorMode,
+    #[kvp(bare)]
+    door_close_mode: DoorMode,
+}
+
+#[derive(Debug, Clone, PartialEq, FromKVPValueEnumNumbers)]
+pub enum AtsAvailable {
+    #[kvp(default, index = "-1")]
+    Neither,
+    AtsSn,
+    Both,
+}
+
+#[derive(Debug, Clone, PartialEq, FromKVPValueEnumNumbers)]
+pub enum AtcAvailable {
+    #[kvp(default)]
+    Unavailable,
+    Manual,
+    Automatic,
+}
+
+#[derive(Debug, Clone, PartialEq, FromKVPValueEnumNumbers)]
+pub enum EbAvailable {
+    #[kvp(default)]
+    Unavailable,
+    Available,
+}
+
+#[derive(Debug, Clone, PartialEq, FromKVPValueEnumNumbers)]
+pub enum ConstSpeedAvailable {
+    #[kvp(default)]
+    Unavailable,
+    Available,
+}
+
+#[derive(Debug, Clone, PartialEq, FromKVPValueEnumNumbers)]
+pub enum HoldBrakeAvailable {
+    #[kvp(default)]
+    Unavailable,
+    Available,
+}
+
+#[derive(Debug, Clone, PartialEq, FromKVPValueEnumNumbers)]
+pub enum ReAdhesionDeviceType {
+    #[kvp(default, index = "-1")]
+    Unavailable,
+    /// Cuts off power instantly and rebuilds it up fast in steps.
+    TypeA,
```

Comment:
Is this their actual names? Ugh engineers shouldn't be allowed to name things

Reply:
Yerp